### DISCUSSION TITLE
Impossible to have a model property be a map from number-like strings to things and to give it an @example?

### DISCUSSION THREAD

**Discussion by fiendish** [microsoft/typespec#5266] [Q&A]
Say I want to have a model property be a map from numbers to other things. Presumably that means myProp is type Record, like e.g.

```
model ModelA {
   myProp: Record<string>;
}
```

But I'm specifically looking for number-alike keys, such as
```
{
   myProp: {"0": "foo", "2": "bar"}
}
```

But it seems like that's impossible? Or it's impossible to give it an `@example` because the syntax disallows the example keys being provided as strings and integers are not valid property names?

Kinda likewise it seems impossible to declare an interface model that just has numeric keys? Like, how can I declare and @example an interface model that receives 
```
{
    "0": "stuff"
}
```
?

**Comment by jchoca** [comment]
I would also like an answer for this. I just want to put other characters in the key for the Record example, e.g.,

```
model Test {
  headers?: Record<string> = #{ "custom-header": "value"};
}
```

but that results in an error.
